pkg/resources/envoy/configmap.go

```diff
@@ -36,12 +36,21 @@ import (
 	kafkautils "github.com/banzaicloud/kafka-operator/pkg/util/kafka"
 )
 
-func (r *Reconciler) configMap(log logr.Logger, extListener v1beta1.ExternalListenerConfig) runtime.Object {
+func (r *Reconciler) configMap(log logr.Logger, extListener v1beta1.ExternalListenerConfig,
+	_ v1beta1.IngressConfig, ingressConfigName, defaultIngressConfigName string) runtime.Object {
+	var configMapName string
+	if ingressConfigName == util.IngressConfigGlobalName {
+		configMapName = fmt.Sprintf(envoyVolumeAndConfigName, extListener.Name, r.KafkaCluster.GetName())
+	} else {
+		configMapName = fmt.Sprintf(envoyVolumeAndConfigNameWithScope, extListener.Name,
+			ingressConfigName, r.KafkaCluster.GetName())
+	}
 	configMap := &corev1.ConfigMap{
 		ObjectMeta: templates.ObjectMeta(
-			fmt.Sprintf(envoyVolumeAndConfigName, extListener.Name, r.KafkaCluster.GetName()),
-			labelsForEnvoyIngress(r.KafkaCluster.GetName(), extListener.Name), r.KafkaCluster),
-		Data: map[string]string{"envoy.yaml": GenerateEnvoyConfig(r.KafkaCluster, extListener, log)},
+			configMapName,
+			labelsForEnvoyIngress(r.KafkaCluster.GetName(), annotationName), r.KafkaCluster),
```

Comment:
Why `annotationName` is passed into `labelsForEnvoyIngress` instead of `extListener.Name`?